lib/ramble/docs/tutorials/using_zips_and_matrices.rst

```diff
@@ -0,0 +1,354 @@
+   https://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+   <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
+   option. This file may not be copied, modified, or distributed
+   except according to those terms.
+
+.. _zips_and_matrices_tutorial:
+
+==============================
+7) Zips and Matrices
+==============================
+
+In this tutorial, you will learn how to generate a more comprehensive set of
+experiments using zips and matrices for
+`WRF <https://www.mmm.ucar.edu/models/wrf>`_, a free and open-source
+application for atmospheric research and operational forecasting applications.
+
+This tutorial builds off of concepts introduced in previous tutorials. Please
+make sure you review those before starting with this tutorial's content.
+
+Create a Workspace
+------------------
+
+To begin with, you need a workspace to configure the experiments. This can be
+created with the following command:
+
+.. code-block:: console
+
+    $ ramble workspace create zips_and_matrices_wrf
+
+
+Activate the Workspace
+----------------------
+
+Several of Ramble's commands require an activated workspace to function
+properly. Activate the newly created workspace using the following command:
+(NOTE: you only need to run this if you do not currently have the workspace
+active).
+
+.. code-block:: console
+
+    $ ramble workspace activate zips_and_matrices_wrf
+
+Configure Experiment Definitions
+--------------------------------
+
+To being with, you need to configure the workspace. The workspace's root
+location can be seen under the ``Location`` output of:
+
+.. code-block:: console
+
+    $ ramble workspace info
+
+Additionally, the files can be edited directly with:
+
+.. code-block:: console
+
+    $ ramble workspace edit
+
+Within the ``ramble.yaml`` file, write the following contents, which are the
+final configuration from the previous tutorial.
+
+.. code-block:: YAML
+
+    ramble:
+      variables:
+        processes_per_node: 4
+        n_ranks: '{processes_per_node}*{n_nodes}'
+        batch_submit: '{execute_experiment}'
+        mpi_command: mpirun -n {n_ranks}
+      applications:
+        wrfv4:
+          workloads:
+            CONUS_12km:
+              experiments:
+                scaling_{n_nodes}:
+                  variables:
+                    n_nodes: [1, 2, 4]
+      spack:
+        concretized: true
+        packages:
+          gcc9:
+            spack_spec: gcc@9.3.0
+          intel-mpi:
+            spack_spec: intel-mpi@2018.4.274
+            compiler: gcc9
+          wrfv4:
+            spack_spec: wrf@4.2 build_type=dm+sm compile_type=em_real nesting=basic ~chem
+              ~pnetcdf
+            compiler: gcc9
+        environments:
+          wrfv4:
+            packages:
+            - intel-mpi
+            - wrfv4
+
+The above configuration will execute 3 experiments, comprising a basic scaling
+study on three different sets of nodes.
```

Comment:
Is this assuming too much of the user to realize the 3 comes from the vector? should we call it out?

Reply:
No, I'm assuming they know this from a previous tutorial already. I can call it out if that's important, but that's not really the purpose of this tutorial.